From 8bdded9cae8f97f9a9b1fe94aef6988be3b7dab4 Mon Sep 17 00:00:00 2001
From: john vincent <john.vincent.xa@bp.renesas.com>
Date: Wed, 1 Mar 2023 18:41:31 +0000
Subject: [PATCH 1/3] Serial flash capability enable for highhope RZ/G2M

spi flash device tree entry details updated for hihope rzg2m board
Reset device register for RPC serial flash control
Enable serial flash commands
Add rzg2m device entry list on the list of supported SPI devices

Signed-off-by: john vincent <john.vincent.xa@bp.renesas.com>
---
 arch/arm/dts/r8a774a1.dtsi              | 17 ++++++++++++++++-
 board/hoperun/hihope-rzg2/hihope-rzg2.c |  8 ++++++++
 configs/hihope_rzg2_defconfig           |  1 +
 drivers/spi/renesas_rpc_spi.c           |  1 +
 4 files changed, 26 insertions(+), 1 deletion(-)

diff --git a/arch/arm/dts/r8a774a1.dtsi b/arch/arm/dts/r8a774a1.dtsi
index 50f453b..47679c1 100644
--- a/arch/arm/dts/r8a774a1.dtsi
+++ b/arch/arm/dts/r8a774a1.dtsi
@@ -26,6 +26,7 @@
 		i2c5 = &i2c5;
 		i2c6 = &i2c6;
 		i2c7 = &i2c_dvfs;
+		spi0 = &rpc;
 	};
 
 	/*
@@ -2377,7 +2378,21 @@
 			resets = <&cpg 917>;
 			#address-cells = <1>;
 			#size-cells = <0>;
-			status = "disabled";
+
+			num-cs = <1>;
+			status = "okay";
+			spi-max-frequency = <40000000>;
+
+			flash0: spi-flash@0 {
+				#address-cells = <1>;
+				#size-cells = <1>;
+				compatible = "spi-flash", "jedec,spi-nor";
+				spi-max-frequency = <40000000>;
+				spi-tx-bus-width = <1>;
+				spi-rx-bus-width = <1>;
+				reg = <0>;
+				status = "okay";
+			};
 		};
 
 		gic: interrupt-controller@f1010000 {
diff --git a/board/hoperun/hihope-rzg2/hihope-rzg2.c b/board/hoperun/hihope-rzg2/hihope-rzg2.c
index 58ab178..3000b47 100644
--- a/board/hoperun/hihope-rzg2/hihope-rzg2.c
+++ b/board/hoperun/hihope-rzg2/hihope-rzg2.c
@@ -50,6 +50,8 @@ DECLARE_GLOBAL_DATA_PTR;
 #define GPIO_WLAN_REG_ON		157
 #define GPIO_BT_REG_ON			158
 
+#define RPC_CMNCR_REGISTER				(0xEE200000)
+
 static int board_rev;
 
 void clear_wlan_bt_reg_on(void)
@@ -149,6 +151,12 @@ int board_init(void)
 
 	clear_wlan_bt_reg_on();
 
+	/*BSZ = 0: Serial flash memory x 1
+	  IO0FV= 11 Output value Hi-Z
+	  MOIIO0/3 = 11 Hi-Z
+	  MD = Manual mode */
+	*(volatile u32 *)(RPC_CMNCR_REGISTER) = 0x00FFF300;
+
 	return 0;
 }
 
diff --git a/configs/hihope_rzg2_defconfig b/configs/hihope_rzg2_defconfig
index d3dd8b1..b5afcb3 100644
--- a/configs/hihope_rzg2_defconfig
+++ b/configs/hihope_rzg2_defconfig
@@ -114,3 +114,4 @@ CONFIG_PARTITION_UUIDS=y
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_NET_RANDOM_ETHADDR=y
 CONFIG_CMD_RTC=y
+CONFIG_CMD_SF=y
diff --git a/drivers/spi/renesas_rpc_spi.c b/drivers/spi/renesas_rpc_spi.c
index 26b6aa8..9122558 100644
--- a/drivers/spi/renesas_rpc_spi.c
+++ b/drivers/spi/renesas_rpc_spi.c
@@ -456,6 +456,7 @@ static const struct udevice_id rpc_spi_ids[] = {
 	{ .compatible = "renesas,rpc-r8a77970" },
 	{ .compatible = "renesas,rpc-r8a77995" },
 	{ .compatible = "renesas,rcar-gen3-rpc" },
+	{ .compatible = "renesas,r8a774a1-rpc-if" },
 	{ }
 };
 
-- 
2.7.4

